Add endpoint to get all users with a specific rank

draeeeeeeeeeeeem opened the pull request:
I haven't tested this yet as I haven't gotten the API setup on a server, although I'm pretty confident it'll work.

draeeeeeeeeeeeem commented:
Regarding [this suggestion](https://github.com/WarzoneMC/Website/issues/23), can be utilized to add a staff page.

jellz commented:
why are you returning 401

jellz commented on routes/minecraft/ranks.js:
```diff
@@ -12,6 +12,20 @@ module.exports = function (app) {
+            var rank = ranks[0];
+            
+            if (!rank) {
+                res.status(401).json({ notFound: true });
```
Why are you returning 401?

draeeeeeeeeeeeem replied:
im copying the other endpoints, the one just below it uses 401 for notfound

MattBSG replied:
HTTP 401 is very different from HTTP 404.

HTTP 401: Unauthorized - you do not have the proper permission to view a resource
HTTP 404: Not Found - the resource is unavailable or does not exist

I’ll open a PR later today to bring the rest up to code and uniform. You should use 404 here

jellz replied:
Exactly what I thought! Thank you. Someone needs to teach the API developers a lesson HTTP, haha.

draeeeeeeeeeeeem commented:
Updated: switched 401 to 404 in accordance with [Matt's PR](https://github.com/WarzoneMC/API/pull/8). Only reason I originally used 401 was to keep style with the rest of the code, which was also using 401.

draeeeeeeeeeeeem commented:
Tested code on local version and can confirm it works.

lukechatton commented:
Endpoint should more clearly show that it's a query by rank.

draeeeeeeeeeeeem commented:
Yeah I realized that, on my local version I've updated
